benchmarks/run_benchmark.py

```diff
@@ -0,0 +1,118 @@
+import sys
+import logging
+import subprocess
+
+import torchvision
+import torch
+import transformers
+from torchvision.prototype.datasets import load as loadpipe
+import torch.nn.functional as F
+from torchvision import transforms
+import time
+import torch.optim as optim
+import torch.profiler
+
+# Relative imports
+from args import arg_parser
+from benchmarks.datasets import prepare_gtsrb_dataset
+from utils import init_fn
+from datasets import prepare_gtsrb_datapipe
+from trainers import train
+from report import create_report
+
+logging.basicConfig(filename='example.log', level=logging.DEBUG)
+
+
+dataset, ispipe, model_name, batch_size, device, num_epochs, num_workers, shuffle, dataloaderv = arg_parser()
+
+if device.startswith("cuda"):
+    nvidiasmi = subprocess.check_output("nvidia-smi", shell=True, text=True)
+    logging.debug(nvidiasmi)
+
+lscpu = subprocess.check_output("lscpu", shell=True, text=True)
+logging.debug(lscpu)
+
+
+if dataloaderv == 1:
+    from torch.utils.data import DataLoader
+elif dataloaderv == 2:
+    from torch.utils.data.dataloader_experimental import DataLoader2 as DataLoader
+else:
+    raise(f"dataloaderv{dataloaderv} is not a valid option")
+
+# Download model
+model_map = {
+    "resnext50_32x4d": torchvision.models.resnext50_32x4d,
+    "mobilenet_v3_large" : torchvision.models.mobilenet_v3_large,
+    "transformerencoder" : torch.nn.TransformerEncoder,
+    "bert-base" : transformers.BertModel,
+
+}
+
+model = model_map[model_name]().to(torch.device(device))
+
+# setup data pipe
+if dataset == "gtsrb":
+    if ispipe:
+        dp = loadpipe(dataset, split="train")
+        logging.debug(f"data format before preprocessing is {next(iter(dp))}")
+
+        dp = prepare_gtsrb_datapipe(batch_size, device, dp)
+        logging.debug(f"data format after preprocessing is \n {next(iter(dp))}\n")
+
+    else:
+        # No further preprocessing needed this returns a tuple of Images and labels as ints
+        # Do I need to do batching and collation manually?
```

Comment:
Nicolas is right that you can do that for `DataLoader`.

I would think it is better to use `.batch` since it will be necessary for `DataLoaderV2`. Then you can pass the same DataPipe to both versions of DL without more changes later.